### PR TITLE
Use quantize export arg in compare pages

### PR DESCRIPTION
**Pull request by glenn-jocher**
Updates the TensorRT export snippets on the model-comparison pages to use the new unified `quantize` export arg.

- `model.export(format="engine", half=True)` → `model.export(format="engine", quantize=16)` (FP16) across 8 compare pages.

The `quantize` arg comes from ultralytics/ultralytics#24918, which replaces the separate `half`/`int8` precision booleans with a single `quantize` value (`16` = FP16, `8` = INT8, `32` = FP32). This keeps the docs aligned with the new API. It should land after that PR ships the `quantize` arg.

I have read the CLA Document and I sign the CLA

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
This PR updates multiple model comparison docs to use the newer `quantize=16` TensorRT export syntax instead of `half=True`, keeping Ultralytics export examples consistent and up to date 🚀

### 📊 Key Changes
- Replaced `model.export(format="engine", half=True)` with `model.export(format="engine", quantize=16)` across several comparison pages.
- Updated TensorRT export examples in docs comparing:
  - PP-YOLOE vs RT-DETR
  - YOLOv10 vs DAMO-YOLO
  - YOLOv10 vs EfficientDet
  - YOLOv10 vs PP-YOLOE
  - YOLOv8 vs YOLOv9
  - YOLOv9 vs YOLOv6
  - YOLOX vs YOLOv10
  - YOLOX vs YOLOv9
- Kept the surrounding training, inference, and deployment examples unchanged, focusing only on export argument modernization 🛠️

### 🎯 Purpose & Impact
- Improves documentation accuracy by aligning examples with the current Ultralytics export API 📘
- Reduces confusion for users copying deployment commands from the docs, especially for TensorRT workflows ✅
- Helps maintain consistency across comparison articles, making the docs easier to trust and follow 🤝
- Supports smoother deployment onboarding for both new and experienced users targeting optimized edge or production inference ⚡

### CODE REVIEW

File: docs/en/compare/yolov10-vs-damo-yolo.md [hunk 1]
```diff
@@ -58,7 +58,7 @@ results = model.train(data="coco8.yaml", epochs=100, imgsz=640)
 results = model("https://ultralytics.com/images/bus.jpg")
 
 # Export the model to TensorRT format
-model.export(format="engine", half=True)
+model.export(format="engine", quantize=16)
```
Comment (UltralyticsAssistant) [inline docs/en/compare/yolov10-vs-damo-yolo.md:61]: ⚠️ **HIGH**: `quantize` does not appear to be a supported export argument in the current Ultralytics API. The export config/docs still expose `half` for FP16 and `int8` for INT8, and unsupported export args are rejected during config validation. Because this is a copy-paste example and the same replacement was made across all 8 pages, users are likely to hit a broken command here. If the intent is 16-bit TensorRT export, this should stay `half=True`. ([github.com](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/engine/exporter.py))

**Suggested change:**
```suggestion
model.export(format="engine", half=True)
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

The change is mechanically consistent, but the new `quantize=16` examples do not match the current Ultralytics export API. The supported precision flags remain `half` and `int8`, and unknown export args are rejected by config validation, so this should be fixed before merge. ([github.com](https://github.com/ultralytics/ultralytics/blob/main/ultralytics/engine/exporter.py))

💬 Posted 1 inline comment
